raytracing/__main__.py

```diff
@@ -1,430 +1,123 @@
-    cavity.append(Space(d=160))
 
-    # Calculate all self-replicating modes (i.e. eigenmodes)
-    (q1,q2) = cavity.eigenModes()
-    print(q1,q2)
 
-    # Obtain all physical (i.e. finite) self-replicating modes
-    qs = cavity.laserModes()
-    for q in qs:
-        print(q)
+class ArgumentsManager:
+    def __init__(self, arguments):
+        self.arguments = arguments
+        self.exampleDirPath = os.path.dirname(os.path.realpath(__file__)) + "\..\examples\\argsExamples"
+
+        self.selectedFileIndex = 0
+        self.htmlTemporaryExampleFile = None
+        self.HTMLDescription = None
+        self.figureObject = None
+        self.base64Figure = None
+        self.selectedFile = None
+        self.HTMLFigure = None
+
+        self.formatter = pygments.formatters.get_formatter_by_name('html',full=True, linenos=True, cssclass="source", style='default')
+        self.lexer = pygments.lexers.get_lexer_by_name('python', stripall=True)
+
+        self.parseArguments()
+
+    def exampleCarousel(self, exampleIndexList):
+        for i in exampleIndexList:
+            self.selectedFileIndex = i
+            self.selectedFile = self.exampleDirPath + '\example{}.py'.format(i)
+            self.generateExample()
+            self.showExample()
+
+    def showExample(self):
+        webbrowser.open("file://" + self.htmlTemporaryExampleFile)
+
+    def generateExample(self):
+        self.createTemporaryHTML()
+        self.generateFigureFromCode()
+        self.generateHTMLHighlightedCode()
+        self.generateHTMLDescription()
+        self.mergeComponentsToHTML()
+
+    def mergeComponentsToHTML(self):
+        with open(self.htmlTemporaryExampleFile, 'a+') as f:
+            f.write(self.HTMLDescription)
+            f.write(self.HTMLFigure)
+            f.write(self.HTMLHighlightedCode)
+            f.close()
+
+    def createTemporaryHTML(self):
+        temp = tempfile.NamedTemporaryFile(delete=False)
+        path = temp.name + '.html'
+        with open(path, 'w') as f:
+            f.write(''''<head><link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous"></head>''')
+
+        self.htmlTemporaryExampleFile = path
+
+    def generateFigureFromCode(self):
+        self.runExampleCode()
+        fig, axes = plt.subplots(figsize=(7, 5))
+        self.figureObject.createRayTracePlot(axes=axes)
+        with tempfile.TemporaryFile(suffix=".png") as tmpfile:
+            plt.savefig(tmpfile, format="png")
+            tmpfile.seek(0)
+            self.base64Figure = base64.b64encode(tmpfile.read()).decode()
+        self.HTMLFigure = ''''<div style="display:inline-block; vertical-align:top"><img class="icon" src="data:image/png;base64,{} "></div>'''.format(
+            self.base64Figure)
+
+    def generateHTMLHighlightedCode(self, code=""):
```

Comment:
Do you plan on using this `code` argument ?

Reply:
It was for future proofing. Daniel talked about inline string code input. Even though i'm not sure about that, I kept it in mind that the argument of -s might be a string to highlight

Reply:
Ok. I guess this could enable a user to create it's own html examples...  in the future